src/meta/src/barrier/mod.rs

```diff
@@ -762,17 +817,33 @@ impl GlobalBarrierManager {
                         _ => {}
                     }
                 }
+                complete_result = self
+                    .completing_task
+                    .next_completed_barrier(
```

Comment:
Actually both the two methods are still there. The name of `next_complete_barrier_response` may be confusing, because it actually means the collect barrier response. The reason for the naming previously is because the name of the protobuf message of the response type is `CompleteBarrierResponse`. I have changed the name to `next_collect_barrier_response`.

Reply:
Sorry for the confusion. What I was asking here is not about the naming, but about the priority of the select arm. Prior to this PR, we poll `next_completed_barrier` first and then `next_complete_barrier_response` but after this PR we switch the order.

Reply:
In this PR, the `next_completed_barrier` takes the mutable reference of `control_stream_manager` to generate the future (though the mutable reference is used, the generated future won't capture the mutable reference).

In the original poll order, the mutable reference of `control_stream_manager` will have been captured by the future generated from `next_complete_barrier_response `, and cannot be used when generating the future of `next_completed_barrier `.